plan/logical_plans.go

```diff
@@ -346,6 +349,45 @@ func (ds *DataSource) deriveTablePathStats(path *accessPath) error {
 		return nil
 	}
 	path.accessConds, path.tableFilters = ranger.DetachCondsForTableRange(ds.ctx, ds.pushedDownConds, pkCol)
+	// If there's no access cond, we try to find that whether there's expression containing correlated column that can be used
+	// for accessing data.
+	if len(path.accessConds) == 0 {
+		for i, filter := range path.tableFilters {
+			eqFunc, ok := filter.(*expression.ScalarFunction)
+			if !ok || eqFunc.FuncName.L != ast.EQ {
+				continue
+			}
+			lCol, lOk := eqFunc.GetArgs()[0].(*expression.Column)
+			if lOk && lCol.Equal(ds.ctx, pkCol) {
+				_, rOk := eqFunc.GetArgs()[1].(*expression.CorrelatedColumn)
+				if rOk {
+					path.accessConds = append(path.accessConds, filter)
+					path.tableFilters = append(path.tableFilters[:i], path.tableFilters[i+1:]...)
+					path.corColInAccessConds = true
+					break
+				}
+			}
+			rCol, rOk := eqFunc.GetArgs()[1].(*expression.Column)
+			if rOk && rCol.Equal(ds.ctx, pkCol) {
+				_, lOk := eqFunc.GetArgs()[0].(*expression.CorrelatedColumn)
+				if lOk {
+					path.accessConds = append(path.accessConds, filter)
+					path.tableFilters = append(path.tableFilters[:i], path.tableFilters[i+1:]...)
+					path.corColInAccessConds = true
+					break
+				}
+			}
+		}
+	}
+	if path.corColInAccessConds {
```

Comment:
Seems that this value is only used in this function, so why make it a member of `accessPath`?